### PR TITLE
Compatibility with geopandas 1.0 and dask-geopandas 0.4.0

### PR DESCRIPTION
**Pull request by hoxbro**
A minor compatibility change to support `dask_geopandas.dask_expr`. 

Removed world example as that dataset is no longer available. Haven't looked at the surrounding text.

I also removed the numpy2 feature, as the original test now pulls it in.

### CODE REVIEW

**Comment by droumis** [comment]
started reviewing the notebook, but got stuck debugging the following:

```python
ddf = dd.from_pandas(sgeodf, npartitions=2).pack_partitions(npartitions=100).persist()

tf.shade(cvs.polygons(ddf, geometry='geometry', agg=ds.mean('population')), cmap=cc.kg)
```

```bash
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File ~/opt/miniconda3/envs/datashader-dev/lib/python3.12/site-packages/dask_expr/_core.py:467, in Expr.__getattr__(self, key)
    466 try:
--> 467     return object.__getattribute__(self, key)
    468 except AttributeError as err:

AttributeError: 'FromPandas' object has no attribute 'pack_partitions'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
File ~/opt/miniconda3/envs/datashader-dev/lib/python3.12/site-packages/dask_expr/_collection.py:619, in FrameBase.__getattr__(self, key)
    616 try:
    617     # Fall back to `expr` API
    618     # (Making sure to convert to/from Expr)
--> 619     val = getattr(self.expr, key)
    620     if callable(val):

File ~/opt/miniconda3/envs/datashader-dev/lib/python3.12/site-packages/dask_expr/_core.py:488, in Expr.__getattr__(self, key)
    487 link = "https://github.com/dask-contrib/dask-expr/blob/main/README.md#api-coverage"
--> 488 raise AttributeError(
    489     f"{err}\n\n"
    490     "This often means that you are attempting to use an unsupported "
    491     f"API function. Current API coverage is documented here: {link}."
    492 )

AttributeError: 'FromPandas' object has no attribute 'pack_partitions'

This often means that you are attempting to use an unsupported API function. Current API coverage is documented here: https://github.com/dask-contrib/dask-expr/blob/main/README.md#api-coverage.

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
Cell In[36], line 1
----> 1 ddf = dd.from_pandas(sgeodf, npartitions=2).pack_partitions(npartitions=100).persist()
      3 tf.shade(cvs.polygons(ddf, geometry='geometry', agg=ds.mean('population')), cmap=cc.kg)

File ~/opt/miniconda3/envs/datashader-dev/lib/python3.12/site-packages/dask_expr/_collection.py:3049, in DataFrame.__getattr__(self, key)
   3046     raise err
   3047 except AttributeError:
   3048     # Fall back to `BaseFrame.__getattr__`
-> 3049     return super().__getattr__(key)

File ~/opt/miniconda3/envs/datashader-dev/lib/python3.12/site-packages/dask_expr/_collection.py:625, in FrameBase.__getattr__(self, key)
    622     return val
    623 except AttributeError:
    624     # Raise original error
--> 625     raise err

File ~/opt/miniconda3/envs/datashader-dev/lib/python3.12/site-packages/dask_expr/_collection.py:614, in FrameBase.__getattr__(self, key)
    611 def __getattr__(self, key):
    612     try:
    613         # Prioritize `FrameBase` attributes
--> 614         return object.__getattribute__(self, key)
    615     except AttributeError as err:
    616         try:
    617             # Fall back to `expr` API
    618             # (Making sure to convert to/from Expr)

AttributeError: 'DataFrame' object has no attribute 'pack_partitions'
```

**Comment by hoxbro** [comment]
> started reviewing the notebook, but got stuck debugging the following:

Forgot to tell you, but spatialpandas does _not_ work with dask-expr. So you have to disable it: `dask config set dataframe.query-planning False` or system environment variable `export DASK_DATAFRAME__QUERY_PLANNING=False`

**Comment by droumis** [comment]
@jbednar Does this match your understanding of interrelationships?

```mermaid
graph TD;
    subgraph SpatialPandas
        A[Polygon]:::spatialpandas -->|used in| E[PolygonArray]:::spatialpandas
        C[MultiPolygon]:::spatialpandas -->|used in| G[MultiPolygonArray]:::spatialpandas
        E -->|used in| F[GeoDataFrame]:::spatialpandas
        G -->|used in| F
        E -->|useable in| N[Pandas DataFrame]:::pandas
        G -->|useable in| N
        A -->|converts to| B[shapely.geometry.Polygon]:::shapely
        C -->|converts to| D[shapely.geometry.MultiPolygon]:::shapely
        B -->|converts to| A
        D -->|converts to| C
    end

    subgraph GeoPandas
        H[GeoSeries]:::geopandas -->|converts to| I[spatialpandas.GeoSeries]:::spatialpandas
        H -->|converts to| J[GeometryArray]:::geopandas
    end

    subgraph Dask
        F -->|used in| L[DataFrame with Geometry]:::dask
        I -->|used in| L
    end

    subgraph Datashader
        M[Datashader]:::datashader
        F -->|usable by| M
        L -->|usable by| M
        J -->|usable by| M
        E -->|usable by| M
        G -->|usable by| M
        N -->|usable by| M
    end

    classDef spatialpandas fill:#7aa9e6,stroke:#000,stroke-width:1px;
    classDef shapely fill:#ff7f0e,stroke:#000,stroke-width:1px;
    classDef geopandas fill:#2ca02c,stroke:#000,stroke-width:1px;
    classDef dask fill:#d62728,stroke:#000,stroke-width:1px;
    classDef datashader fill:#9467bd,stroke:#000,stroke-width:1px;
    classDef pandas fill:#bcbd22,stroke:#000,stroke-width:1px;
```

**Comment by droumis** [comment]
Changed some of the text to make it make more sense (to me at least). I also replaced the dataset with one that shows something more recognizable (USA! USA!)

image

image

File: tox.ini [hunk 1]
```diff
@@ -25,7 +25,7 @@ commands = pytest datashader -k "not test_tiles" -n logical --dist loadgroup --c
 [_examples]
 description = Test that default examples run
 deps = .[examples, tests]
-commands = pytest -n logical --dist loadscope --nbval-lax examples --benchmark-skip
+commands = pytest -n logical --dist loadscope --nbval-lax examples --benchmark-skip -v
```
Comment (hoxbro) [inline tox.ini:28]:
```suggestion
commands = pytest -n logical --dist loadscope --nbval-lax examples --benchmark-skip
```

**Comment by droumis** [comment]
Working on a updated version:

```mermaid
graph TD;
    subgraph SpatialPandas
        A[Polygon]:::spatialpandas -->|used in| E[PolygonArray]:::spatialpandas
        C[MultiPolygon]:::spatialpandas -->|used in| G[MultiPolygonArray]:::spatialpandas
        E -->|used in| F[GeoDataFrame]:::spatialpandas
        G -->|used in| F
        I[GeoSeries]:::spatialpandas
    end

    subgraph GeoPandas
        H[GeoSeries]:::geopandas -->|converts to| I
        H -->|used in| J[GeoDataFrame]:::geopandas
    end

    subgraph Pandas
        E -->|used in| N[DataFrame with Geometry]:::pandas
        G -->|used in| N
    end

    subgraph Shapely
        A -->|converts to| B[shapely.geometry.Polygon]:::shapely
        C -->|converts to| D[shapely.geometry.MultiPolygon]:::shapely
        B -->|converts to| A
        D -->|converts to| C
    end

    subgraph Dask
        F -->|used in| L[DataFrame with Geometry]:::dask
        I -->|used in| L
    end

    subgraph Datashader
        M[Datashader]:::datashader
        F -->|usable by| M
        L -->|usable by| M
        J -->|usable by| M
        E -->|usable by| M
        G -->|usable by| M
        N -->|usable by| M
    end

    classDef spatialpandas fill:#7aa9e6,stroke:#000,stroke-width:1px;
    classDef shapely fill:#ff7f0e,stroke:#000,stroke-width:1px;
    classDef geopandas fill:#2ca02c,stroke:#000,stroke-width:1px;
    classDef dask fill:#d62728,stroke:#000,stroke-width:1px;
    classDef datashader fill:#9467bd,stroke:#000,stroke-width:1px;
    classDef pandas fill:#bcbd22,stroke:#000,stroke-width:1px;

    subgraph Legend
        LA[SpatialPandas]:::spatialpandas
        LB[GeoPandas]:::geopandas
        LC[Pandas]:::pandas
        LD[Shapely]:::shapely
        LE[Dask]:::dask
        LF[Datashader]:::datashader
    end
```

**Comment by droumis** [comment]
V3 .. fixing subgraph titles overlapping with arrows with some trickery

```mermaid
graph TD;
    subgraph SP[ ]
        spatialpandas[SpatialPandas]:::spatialpandas_title
        A[Polygon]:::spatialpandas -->|used in| E[PolygonArray]:::spatialpandas
        B[MultiPolygon]:::spatialpandas -->|used in| F[MultiPolygonArray]:::spatialpandas
        E -->|used in| G[GeoDataFrame or GeoSeries]:::spatialpandas
        F -->|used in| G
    end

    subgraph SH[ ]
        shapely[Shapely]:::shapely_title
        H[Polygon]:::shapely <-->|converts to| A
        I[MultiPolygon]:::shapely <-->|converts to| B
    end

    subgraph GP[ ]
        geopandas[GeoPandas]:::geopandas_title
        J[GeoDataFrame or GeoSeries]:::geopandas
    end

    subgraph PD[ ]
        pandas[Pandas]:::pandas_title
        K[DataFrame or Series]:::pandas
    end

    subgraph SPD[ ]
        spatialpandas.dask[SpatialPandas.Dask]:::dask_title
        L[DaskGeoDataFrame or DaskGeoSeries]:::dask
    end

    M{Datashader}:::datashader_title

    F -->|usable in| K
    E -->|usable in| K
    
    G -->|converts to| L

    G <-->|converts to| J

    G -->|usable by| M
    L -->|usable by| M
    J -->|usable by| M
    K -->|usable by| M

    classDef spatialpandas fill:#4e79a7,stroke:#000,stroke-width:1px;
    classDef shapely fill:#f28e2b,stroke:#000,stroke-width:1px;
    classDef geopandas fill:#59a14f,stroke:#000,stroke-width:1px;
    classDef dask fill:#76b7b2,stroke:#000,stroke-width:1px;
    classDef datashader fill:#b07aa1,stroke:#000,stroke-width:1px;
    classDef pandas fill:#edc948,stroke:#000,stroke-width:1px;
    classDef spatialpandas_title fill:#fff,stroke:#4e79a7,stroke-width:8px,color:black;
    classDef shapely_title fill:#fff,stroke:#f28e2b,stroke-width:8px,color:black;
    classDef geopandas_title fill:#fff,stroke:#59a14f,stroke-width:8px,color:black;
    classDef dask_title fill:#fff,stroke:#76b7b2,stroke-width:8px,color:black;
    classDef pandas_title fill:#fff,stroke:#edc948,stroke-width:8px,color:black;
    classDef datashader_title fill:#fff,stroke:#b07aa1,stroke-width:8px,color:black;
```

**Comment by jbednar** [comment]
@droumis , that looks good! Are the double-headed arrows that point _in_ to the Shapely box intentional? I would have thought that Datashader has no interest in converting things to Shapely.

**Comment by droumis** [comment]
They were intentional because the notebook mentions the `to_shapely` method. But I agree that it's a bit of a distraction so I've now removed those lines going back toward shapely.

**Comment by droumis** [comment]
@hoxbro, I think we can merge now

**Review by jbednar — APPROVED** [review]
Looks good, thanks!

File: examples/user_guide/8_Polygons.ipynb [hunk 2]
```diff
@@ -571,6 +664,44 @@
...
+    "\n",
+    "Before running these examples, you will need spatialpandas installed with pip:\n",
+    "\n",
+    "```\n",
+    "$ pip install spatialpandas\n",
+    "```\n",
+    "\n",
+    "or conda:\n",
+    "```\n",
+    "$ conda install -c pyviz/label/dev spatialpandas\n",
```
[the remark this answers is not in the document]
Comment (droumis) [reply examples/user_guide/8_Polygons.ipynb:683]: temporary in order to pull spatialpandas [v0.4.11](https://github.com/holoviz/spatialpandas/releases/tag/v0.4.11a2)

File: examples/user_guide/8_Polygons.ipynb [hunk 3]
```diff
@@ -571,6 +664,44 @@
...
+    "```\n",
+    "\n",
+    "## Configuring Dask for SpatialPandas Compatibility\n",
+    "\n",
+    "SpatialPandas does not work with `dask-expr`. Disable Dask's query planning using one of the following methods:\n",
```
[the remark this answers is not in the document]
Comment (droumis) [reply examples/user_guide/8_Polygons.ipynb:688]: is this sufficient?

Dask version [2024.3.0](https://docs.dask.org/en/stable/changelog.html#:~:text=2024.3.0-,%C2%B6,-Released%20on%20March) made a new 'query planning' (`dask-expr`) version of their `DataFrame` enabled by default.  As of SpatialPandas v0.4.11a2, it does not yet work with `dask-expr`. Until SpatialPandas is updated, you can use one of the following workarounds to work with Dask 2024.3.0 or later:

File: examples/conftest.py [hunk 4]
```diff
@@ -14,3 +16,13 @@
         "user_guide/7_Networks.ipynb",
         "user_guide/8_Polygons.ipynb",
     ]
+
+
+# Will give an exit code 143 on Github Actions
+# First seen in: https://github.com/holoviz/datashader/pull/1347
+# Which tried to update to Geopandas 1.0, but during development
+# Numpy 2 was also able to be solved for on Conda-Forge
```
[the remark this answers is not in the document]
Comment (droumis) [reply examples/conftest.py:24]: I'll leave this to @hoxbro

[the remark this answers is not in the document]
Comment (hoxbro) [reply examples/conftest.py:24]: > Is this specific to GH Actions

Yes, this is why I check for the environment variable. I have opened a new PR where I debug: https://github.com/holoviz/datashader/pull/1348. 

A quick Google search shows this is likely a resource error in that example. I will play around with it in the PR.  

What it was waiting for was time to investigate. Many changes happened during the development of this PR (new dependencies), and I wanted to make a release this week. So it was always meant to be temporary.

[the remark this answers is not in the document]
Comment (hoxbro) [reply examples/conftest.py]: It was a memory issue. I have merged that PR into this PR. The notebook is no longer skipped on Ubuntu.

File: examples/user_guide/8_Polygons.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (hoxbro) [reply examples/user_guide/8_Polygons.ipynb]: By default, Dask version [2024.3.0](https://docs.dask.org/en/stable/changelog.html#:~:text=2024.3.0-,%C2%B6,-Released%20on%20March) enabled a new query-planning (`dask-expr`) version of their DataFrame API. SpatialPandas v0.4.11 does not yet support query-planning. Until SpatialPandas supports query planning, you can use one of the following steps to use the classic Dask DataFrame API:

**Pull request merged**